### PR TITLE
Add mocks docs

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

Add docs for mocks and spies.

## 🖼️ Screenshots

Screenshot 2023-10-01 at 18 06 10

Screenshot 2023-10-01 at 18 06 28

Screenshot 2023-10-01 at 18 06 43

### CODE REVIEW

File: docs/.vitepress/config.ts [hunk 1]
```diff
@@ -51,6 +51,9 @@ export default defineConfig({
       }, {
         text: 'Assertions',
         link: '/assertions'
+      },{
+        text: 'Mocks',
+        link: '/mocks'
```
Comment (Tito-Kati) [inline docs/.vitepress/config.ts:56]:
```suggestion
        text: 'Test doubles',
        link: '/test-doubles'
```

File: docs/mocks.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Mocks
```
Comment (Tito-Kati) [inline docs/mocks.md:1]:
```suggestion
# Test doubles
```

(rename the file too)

File: docs/mocks.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+When creating tests, you might need to override existing function to be able to write isolated tests from external behaviour. To accomplish this, you can use mocks. You can also check that a function was called with certain arguments or even a number of times with a spy.
+
+## mock
```
Comment (Tito-Kati) [inline docs/mocks.md:5]:
```suggestion
## `mock`
```

File: docs/mocks.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+function test_success() {
+  mock ps echo hello world
+  assert_equals "hello world" "$(ps)"
+}
+```
+
+## spy
+> `spy "function"
+
+Override the behaviour of a function
```
Comment (Tito-Kati) [inline docs/mocks.md:21]: What's the digference with a mock?

File: docs/mocks.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+*Example:*
+```bash
+function test_success() {
+  mock ps echo hello world
+  assert_equals "hello world" "$(ps)"
+}
+```
+
+## spy
```
Comment (Tito-Kati) [inline docs/mocks.md:18]:
```suggestion
## `spy`
```

File: docs/.vitepress/config.ts (hunk 1 above)
Comment (Tito-Kati) [reply docs/.vitepress/config.ts:56]: Move this block before assertions

File: docs/mocks.md (hunk 4 above)
Comment (Tito-Kati) [reply docs/mocks.md:21]: Add links to all spy assertions

**Review by Tito-Kati — CHANGES_REQUESTED** [review]
There is no doc for the new spy assertions.

**Review by JesusValera — APPROVED** [review]

File: docs/mocks.md (hunk 4 above)
Comment (Chemaclass) [reply docs/mocks.md:21]: Those 3 functions are displayed inside the spy docs examples. Should we define them as separate functions? 🤔 
```bash
assert_have_been_called
assert_have_been_called_with
assert_have_been_called_times
```

**Review by Tito-Kati — APPROVED** [review]

**Pull request merged**